include/nbl/video/CCUDADevice.h

```diff
@@ -179,16 +179,74 @@ class CCUDADevice : public core::IReferenceCounted
 		static CUresult acquireAndGetMipmappedArray(GraphicsAPIObjLink<video::IGPUImage>* linksBegin, GraphicsAPIObjLink<video::IGPUImage>* linksEnd, CUstream stream);
 		static CUresult acquireAndGetArray(GraphicsAPIObjLink<video::IGPUImage>* linksBegin, GraphicsAPIObjLink<video::IGPUImage>* linksEnd, uint32_t* arrayIndices, uint32_t* mipLevels, CUstream stream);
 #endif
+		CUdevice getInternalObject() const { return m_handle; }
+		const CCUDAHandler* getHandler() const { return m_handler.get();  }
 
+		struct SSharedCUDAMemory : core::IReferenceCounted
```

Comment:
when you inherit from IRferenceCounted you need to make it a `class` and encapsulate all fields (also making as many as possible constant) and **make sure ctor and dtor are protected** (you can declare a friendship with the factory that makes them though)